src/tools/illink/src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs

```diff
@@ -111,17 +113,23 @@ public override void Initialize (AnalysisContext context)
 						return;
 
 					var returnType = method.ReturnType;
-					foreach (var diagnostic in ProcessGenericParameters (returnType, method.Locations[0]))
-						context.ReportDiagnostic (diagnostic);
+					if (method.Locations.Length > 0) {
+						foreach (var diagnostic in ProcessGenericParameters (returnType, method.Locations[0]))
```

Comment:
Looks like it still might be worth defining a new helper for this, since we're repeating the same code quite a bit.

Also -- is skipping the diagnostic in this instance the right behavior? Are we always guaranteed that a diagnostic cannot occur without a location?

Reply:
+1 to not feeling comfortable disabling the code if there's no location. Pretty much any compiler generated code will not have a location. For the most part the analyzer should not see it, but if it does happen, I think we should still analyze it. Unless not having location is defined to mean something in Roslyn... and that meaning would imply that analyzing it is not a good idea.